src/main/java/org/cryptomator/cryptolib/v3/CryptorImpl.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Sebastian Stenzel and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the accompanying LICENSE.txt.
+ *
+ * Contributors:
+ *     Sebastian Stenzel - initial API and implementation
+ *******************************************************************************/
+package org.cryptomator.cryptolib.v3;
+
+import org.cryptomator.cryptolib.api.Cryptor;
+import org.cryptomator.cryptolib.api.Masterkey;
+import org.cryptomator.cryptolib.v1.CryptorProviderImpl;
+
+import java.security.SecureRandom;
+
+class CryptorImpl implements Cryptor {
+
+	private final Masterkey masterkey;
+	private final FileContentCryptorImpl fileContentCryptor;
+	private final FileHeaderCryptorImpl fileHeaderCryptor;
+	private final FileNameCryptorImpl fileNameCryptor;
+
+	/**
+	 * Package-private constructor.
+	 * Use {@link CryptorProviderImpl#provide(Masterkey, SecureRandom)} to obtain a Cryptor instance.
+	 */
+	CryptorImpl(Masterkey masterkey, SecureRandom random) {
```

Comment:
@overheadhunter after decoding `vault.uvf`, the raw `latestFileKey` and the raw `fileKey` are concatenated and to into `masterkey` (corresponding to `enc` and `mac` part)? 

After key rotation, how is the cryptor instantiated as then we could also have a file with previous file key? Is this out of scope in this pr?

Reply:
This is not yet possible using the existing API. Currently, this branch is merely a demonstrator for the adjusted file header.

The API really needs to be extended to accept separate keys.

Reply:
see newly added API:

https://github.com/cryptomator/cryptolib/blob/084e78a213d7fd5f193d8f545cd7a366988ba996/src/main/java/org/cryptomator/cryptolib/api/Cryptor.java#L26-L32